### PR TITLE
adtte finalization

### LINKED ISSUE #5: Program ADTTE

### PR DESCRIPTION
**Pull request by SHAESEN2**
closes #5. Needs an update when {envsetup} is utilized.

### CODE REVIEW

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Thanks a lot for your efforts @SHAESEN2!

YOu may have missed this as you couldn't attend our initial kick off meeting but the plan for this submission pilot was to use as much of the functionality of {admiral} and the other pharmaverse packages as posible and only rely on the {tidyverse} in case some functionality is not covered by these packages. {admiral} does in fact have quite powerful time-to-event functionality as described [here](https://pharmaverse.github.io/admiral/articles/bds_tte.html). Thus, I'd ask you or someone else from the team to adjust your code accordingly. Thank you!

**Comment by SHAESEN2** [comment]
I thought this was a nice-to-have. I prefer to have a code that is more readable and used admiral or tidyverse when necessary. Perhaps we can bring this PR in and then make an issue to make it more align with admiral although I think it is overkill for this simple dataset.

**Comment by wangben718** [comment]
Hi @SHAESEN2 and @thomas-neitmann , is it possible to have a quick call for this issue? I tried to use more admiral functions in the program for ADTTE but I am not very clear about Steven's program. I am available from 8:30 am to 11 pm on weekdays.

**Comment by robertdevine** [comment]
@wangben718 Use of '[reprex](https://cran.r-project.org/web/packages/reprex/index.html)' may be helpful to discuss differing branch codes for alignment to the [recommendation](https://pharmaverse.github.io/admiral/articles/bds_tte.html). Thank you.

**Comment by SHAESEN2** [comment]
Hey @wangben718,

I am happy to help you understand the code, but I will be unable to help you to use more admiral functions. Let me know your email and then I can sent an invite.

**Comment by wangben718** [comment]
> Hey @wangben718,
> 
> I am happy to help you understand the code, but I will be unable to help you to use more admiral functions. Let me know your email and then I can sent an invite.

Great! My email is benjamin.wang@merck.com. Thanks.

**Comment by robertdevine** [comment]
@thomas-neitmann The link you shared successfully creates the time-to-event ADaM using the cdisc pilot 1 core. The derived data set (adtte) does not in every record match Pilot 2 (cf. ICH E9 TEAE specification). Overall, the derived data sets match the resulting tables provided in the [link](https://pharmaverse.github.io/admiral/articles/bds_tte.html) you shared above.

_Observations_:
[1]The lubridate package is needed for the TRTEMFL indicator variable valuation. Regarding the '30' days, the 'x' value in the TRTEMFL formulation is associated (in the specification) to the drug half-life.
Published studies for Xanomeline (from 2002) report drug half-life ~30 hours. 
[2]The AESTDT variable must be derived to complete the time-to-event for the by group examples. That might not be immediately obvious to someone just starting to create BDS time-to-event ADaM.
[3]No RS domain data is provided in the pilot.

Should an attempt be made to align the Pilot 3 adtte to the Pilot 2 adtte data set result before we finalize? This was discussed at the kickoff meeting as a possible verification check. [...._and is feasible if necessary_.] Otherwise just a matter of which columns should be included in the Pilot3 adtte (Pilot2 - 26 columns, Pilot 3 link - 18 columns). 

Very cool to take the time to use Pharmaverse Admiral to derive the ADaM datasets. The documentation is excellent. Thanks.

@SHAESEN2 , @wangben718 , do you plan to finalize the 'official' (Pharmaverse) commit for adtte? If necessary, I can push it here but would want to hear back from @thomas-neitmann and @laxamanaj about alignment to Pilot2 ADaM set since that was discussed earlier as a verification check point. Thanks.

**Comment by wangben718** [comment]
I revise the program for ADTTE to use more function from admiral. Let me know if there is any comment.

**Comment by robertdevine** [comment]
Thank you, @wangben718. As discussed at the weekly **_Check In_**, a visual inspection was performed to 
compare the Pilot3 dataset (ADTTE) with the previous FDA pilots. The match is essentially complete, the Pilot3 
dataset includes (1)labels in the ADTTE dataset and (2)<date> format for TRTSDT, TRTEDT, STARTDT, and ADT
(as noted earlier by @SHAESEN2).

The "submission/datasets/adtte.xpt" file written out by the program is _not compatible_ with FDA. The {xportr}(https://cran.rstudio.com/web/packages/xportr/xportr.pdf)' package provides a wrapper function ('xportr_write')
which writes an xpt v5 transport file to align with the {admiral} [documentation](https://pharmaverse.github.io/admiral/articles/bds_tte.html) and [CDISC](https://www.cdisc.org/). 

Please respond here if you will push that change (or if I should do it). 

Overall, ADTTE dataset looks close to '**Done**'. @thomas-neitmann, @laxamanaj, @bms63

**Review by laxamanaj — APPROVED** [review]
I've reviewed the code for ADTTE and labels now match for STARTDT and SRCDOM.  I've also updated the call to ADSL and ADAE to now use the current Pilot 3 generated data and the diff still came out the same.  I also updated the generation of adtte.xpt, which was was overwriting adadas.xpt.

I'm going to push these minor updates to the adtte  branch, then we should be good to merge to main after the auto-checks are complete.

As far as the diff goes, since these are all formats and attributes and are not showing up as issues in the P21 run, I'm ok to let this go for this pilot.  We can also note these in the QC findings wiki.

**Review by laxamanaj — APPROVED** [review]
approved to get to main in order to run KM plot. but will need to clean up code a bit more.

File: submission/programs/adtte_bw.R [hunk 1]
```diff
@@ -0,0 +1,175 @@
+###########################################################################
+#' developpers : Steven Haesendonckx/Bingjun Wang
+#' date: 13NOV2022
+#' modification History:
+#' 1. Try to use admiral to generate ADTTE
```
[the remark this answers is not in the document]
Comment (laxamanaj) [reply submission/programs/adtte_bw.R]: n/a adtte_bw.R not used.

File: submission/programs/adtte_bw.R [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+library(haven)
+library(admiral)
+library(dplyr)
+
+# read source -------------------------------------------------------------
+
+adsl <- haven::read_xpt(file.path("adam", "adsl.xpt"))
+adae <- haven::read_xpt(file.path("adam", "adae.xpt"))
+ds <- haven::read_xpt(file.path("sdtm", "ds.xpt"))
```
[the remark this answers is not in the document]
Comment (laxamanaj) [reply submission/programs/adtte_bw.R]: n/a adtte_bw.R not used.

File: submission/programs/adtte_bw.R [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+labsupdated <- sapply(colnames(adtte), FUN = function(x) attr(adtte[[x]], "label"))
+labsupdated[unlist(lapply(labsupdated,is.null))]
+
+# read original ADTTE -----------------------------------------------------
+
+prod <- haven::read_xpt(file.path("adam", "adtte.xpt"))
+labsprod <- sapply(colnames(prod), FUN = function(x) attr(prod[[x]], "label"))
+
+# QC dev vs prod ----------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (laxamanaj) [reply submission/programs/adtte_bw.R]: n/a adtte_bw.R not used.

File: submission/programs/adtte_bw.R [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    new_vars = vars(AGE, AGEGR1, AGEGR1N, RACE, RACEN, SAFFL, SEX, SITEID, TRT01A,
+                    TRT01AN, TRTDURD, TRTEDT, TRT01P, TRTSDT),
+    by_vars = vars(STUDYID, USUBJID)
+) |> 
+  dplyr::rename(TRTA = TRT01A,
+                   TRTAN = TRT01AN,
+                   TRTDUR = TRTDURD,
+                   TRTP = TRT01P)
+
+# Add Labels --------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (laxamanaj) [reply submission/programs/adtte_bw.R]: n/a adtte_bw.R not used.

**Pull request merged**